rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -374,7 +378,13 @@ public Response createPoly(byte[] inputToAutodetectType) {
 
 
         if (spec != null) {
-            return launch(potentialYaml, spec);
+            try {
+                return launch(potentialYaml, spec);
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                log.warn("Failed REST deployment launching "+spec+": "+e);
+                throw WebResourceUtils.badRequest(e, "Error launching blueprint (autodetecting)");
```

Comment:
This seemed like the best status code to use and it's what we use on line 294/297.  Do you think there's a better one?  If we do nothing it will give Internal Server Error which seems worse?